### PR TITLE
Improvement : updating data-cy for e2e tests

### PR DESCRIPTION
**Pull request by justinehell**
- Updating and standardization `data-cy` for e2e tests
- Remove escape char & use single quote for data-cy
- Rename `NewProjectForm` and `NewProjectDialog` to `CreateProjectForm` and `CreateProjectDialog`
- Rename `NewProjectTemplateForm` and `NewProjectTemplateDialog` to `CreateProjectTemplateForm` and `CreateProjectTemplateDialog` 
- Use path instead of label in data-cy for files to avoid problem when multiple files have same name + update e2e tests
- Fix typos

### CODE REVIEW

File: src/components/ModelizerModelView.vue [hunk 1]
```diff
@@ -14,7 +14,7 @@
       <q-page>
         <div
           id="root"
-          data-cy="modelizer-model-view-draw-root"
+          data-cy="draw-root"
```
Comment (Zorin95670) [inline src/components/ModelizerModelView.vue:17]: draw-area

Comment (justinehell) [reply src/components/ModelizerModelView.vue]: done

File: src/components/menu/FileExplorerActionMenu.vue [hunk 2]
```diff
@@ -2,31 +2,37 @@
...
-    data-cy="file-explorer-action-menu"
+    class="file-explorer-menu"
+    data-cy="file-explorer-menu"
   >
     <q-list style="min-width: 150px">
       <template v-if="file.isFolder">
-        <q-item clickable @click="createFile(true)">
+        <q-item
+          clickable
+          data-cy="create-folder"
```
Comment (Zorin95670) [inline src/components/menu/FileExplorerActionMenu.vue:12]: -action

File: src/components/inputs/ObjectInput.vue [hunk 3]
```diff
@@ -33,6 +33,7 @@
           name="fa-solid fa-circle-exclamation"
           size="xs"
           :title="$t('errors.plugin.object')"
+          data-cy="object-attributes-error"
```
Comment (k-y0u) [inline src/components/inputs/ObjectInput.vue:36]: no need to add it if not used

File: src/components/inputs/ObjectInput.vue [hunk 4]
```diff
@@ -41,6 +42,7 @@
       :plugin="plugin"
       :full-name="fullName"
       :current-error="currentError"
+      data-cy="object-attributes-list"
```
Comment (k-y0u) [inline src/components/inputs/ObjectInput.vue:45]: no need to add it if not used

File: src/components/drawer/ComponentDefinitionsDrawer.vue [hunk 5]
```diff
@@ -20,7 +20,7 @@
       <q-input
         clearable
         v-model="definitionFilter"
-        data-cy="filter-plugin-definitions"
+        data-cy="filter-plugin-definitions-input"
```
Comment (k-y0u) [inline src/components/drawer/ComponentDefinitionsDrawer.vue:23]: `definitions-filter-input`

File: src/components/drawer/ComponentDefinitionsDrawer.vue [hunk 6]
```diff
@@ -35,10 +35,10 @@
           header-class="text-bold"
           class="plugin-definitions"
           :key="plugin.data.name"
-          :data-cy="`plugin-definitions-${plugin.data.name}`"
+          :data-cy="`plugin-definitions_${plugin.data.name}`"
```
Comment (k-y0u) [inline src/components/drawer/ComponentDefinitionsDrawer.vue:38]: `component-defnitions-item`

File: src/components/drawer/ComponentDefinitionsDrawer.vue [hunk 7]
```diff
@@ -61,7 +61,7 @@
           data-cy="`template-definitions"
```
Comment (k-y0u) [inline src/components/drawer/ComponentDefinitionsDrawer.vue:61]: `template-definitions-item`

File: src/components/drawer/ComponentDetailPanel.vue [hunk 8]
```diff
@@ -51,6 +51,7 @@
                 v-model="selectedComponentId"
                 class="q-px-md q-pb-sm"
                 :label="$t('plugin.component.attribute.id')"
+                data-cy="selected-component-input"
```
Comment (k-y0u) [inline src/components/drawer/ComponentDetailPanel.vue:54]: no need ?

File: src/components/drawer/ComponentDetailPanel.vue [hunk 9]
```diff
@@ -60,6 +61,7 @@
               <q-checkbox
                 v-model="forceSave"
                 :label="$t('plugin.component.attribute.forceSave')"
+                data-cy="force-save-checkbox"
```
Comment (k-y0u) [inline src/components/drawer/ComponentDetailPanel.vue:64]: no need ?

File: src/components/menu/FileExplorerActionMenu.vue [hunk 10]
```diff
@@ -2,31 +2,37 @@
   <q-menu
     ref="menu"
     @hide="emit('hide:menu')"
-    class="file-explorer-action-menu"
-    data-cy="file-explorer-action-menu"
+    class="file-explorer-menu"
+    data-cy="file-explorer-menu"
```
Comment (k-y0u) [inline src/components/menu/FileExplorerActionMenu.vue:6]: keep `-action`

File: src/components/menu/FileExplorerActionMenu.vue (hunk 2 above)
Comment (k-y0u) [reply src/components/menu/FileExplorerActionMenu.vue:12]: `create-folder-action-item`

File: src/components/menu/GitBranchActionMenu.vue [hunk 11]
```diff
@@ -1,22 +1,22 @@
 <template>
   <q-menu ref="menu">
-    <q-list>
+    <q-list data-cy="git-branch-action-menu">
```
Comment (k-y0u) [inline src/components/menu/GitBranchActionMenu.vue:3]: Move this data-cy to q-menu

File: src/components/menu/GitBranchExpandListMenu.vue
Comment (k-y0u) [inline src/components/menu/GitBranchExpandListMenu.vue:1]: TODO: rename -> it's note a menu

File: src/components/menu/GitBranchExpandListMenu.vue [hunk 12]
```diff
@@ -1,7 +1,7 @@
 <template>
   <q-item clickable class="text-weight-bold text-grey-7">
     <q-item-section avatar/>
-    <q-item-section data-cy="git-branch-expand-menu-text">
+    <q-item-section data-cy="menu-text">
```
Comment (k-y0u) [inline src/components/menu/GitBranchExpandListMenu.vue:4]: `expand-list-message` or `expand-list-text`

File: src/components/menu/GitBranchMenu.vue [hunk 13]
```diff
@@ -23,7 +23,7 @@
           <q-icon
             color="primary"
             name="fa-solid fa-question"
-            data-cy="git-menu-status"
+            data-cy="git-status"
```
Comment (k-y0u) [inline src/components/menu/GitBranchMenu.vue:26]: git-status-icon

Comment (k-y0u) [reply src/components/menu/GitBranchMenu.vue:26]: or `git-status-item` in q-item

File: src/components/menu/GitBranchMenu.vue [hunk 14]
```diff
@@ -23,7 +23,7 @@
...
           />
         </q-item-section>
         <q-item-section>{{ $t('actions.git.status') }}</q-item-section>
```
Comment (k-y0u) [inline src/components/menu/GitBranchMenu.vue:29]: divide

File: src/components/menu/GitBranchMenu.vue [hunk 15]
```diff
@@ -64,22 +64,22 @@
...
-            v-if="showLocal || index < maxItem"
             @action:done="menu.hide()"
-            :data-cy="`git-menu-branch-local-${branch.name}`"
+            :data-cy="`local-branch_${branch.name}`"
           />
         </template>
         <git-branch-expand-list-menu
-          data-cy="git-branch-expand-local-menu"
+          v-if="localBranches.length > maxItem"
+          data-cy="expand-local-branch-menu"
```
Comment (k-y0u) [inline src/components/menu/GitBranchMenu.vue:80]: should be put directly in component

File: src/components/menu/GitBranchMenu.vue [hunk 16]
```diff
@@ -88,22 +88,22 @@
...
-            v-if="showRemote || index < maxItem"
             @action:done="menu.hide()"
-            :data-cy="`git-menu-branch-remote-${branch.name}`"
+            :data-cy="`remote-branch_${branch.name}`"
           />
         </template>
         <git-branch-expand-list-menu
-          data-cy="git-branch-expand-remote-menu"
+          v-if="remoteBranches.length > maxItem"
+          data-cy="expand-remote-branch-menu"
```
Comment (k-y0u) [inline src/components/menu/GitBranchMenu.vue:104]: should be put directly in component

File: src/components/menu/ModelizerSettingsMenu.vue [hunk 17]
```diff
@@ -4,7 +4,7 @@
     color="white"
     text-color="primary"
     icon="fa-solid fa-gear"
-    data-cy="project-settings"
+    data-cy="project-settings-button"
```
Comment (k-y0u) [inline src/components/menu/ModelizerSettingsMenu.vue:7]: `modelizer-settings-button` or rename file

File: src/components/menu/ModelizerSettingsMenu.vue [hunk 18]
```diff
@@ -19,7 +19,7 @@
             v-if="menuItem.visible"
             clickable
             class="settings-item"
-            :data-cy="`git-settings-menu-${menuItem.key}`"
+            :data-cy="`menu_${menuItem.key}`"
```
Comment (k-y0u) [inline src/components/menu/ModelizerSettingsMenu.vue:22]: item_...

File: src/components/ModelizerTextView.vue [hunk 19]
```diff
@@ -10,6 +10,7 @@
         class="q-ml-lg"
         left-label
         :label="$t('page.modelizer.fileExplorer.filterParsableFiles')"
+        data-cy="filter-files-checkbox"
```
Comment (k-y0u) [inline src/components/ModelizerTextView.vue:13]: no need

File: src/components/ModelizerModelView.vue [hunk 20]
```diff
@@ -14,7 +14,7 @@
       <q-page>
         <div
           id="root"
-          data-cy="modelizer-model-view-draw-root"
+          data-cy="draw-area"
```
Comment (k-y0u) [inline src/components/ModelizerModelView.vue:17]: draw-container (div -> container ?)

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: cypress/e2e/HomePage/delete.project.feature [hunk 21]
```diff
@@ -6,71 +6,71 @@ Feature: Test homepage: project deletion
     And  I visit the "/"
 
   Scenario: Delete existing project
-    When I click on "[data-cy=\"new-project\"]"
-    And  I set on "[data-cy=\"new-project-form\"] [data-cy=\"project-name-input\"]" text "projectName"
-    And  I click on "[data-cy=\"new-project-form\"] [data-cy=\"new-project-form-submit\"]"
+    When I click on '[data-cy="new-project"]'
+    And  I set on '[data-cy="new-project-form"] [data-cy="project-name-input"]' text "projectName"
+    And  I click on '[data-cy="new-project-form"] [data-cy="new-project-form-submit"]'
     Then I expect current url is "modelizer/projectName/model"
```
[the remark this answers is not in the document]
Comment (justinehell) [reply cypress/e2e/HomePage/delete.project.feature:12]: done

File: src/components/form/GitAddRemoteForm.vue [hunk 22]
```diff
@@ -17,7 +17,7 @@
       v-model="repository"
       :label="$t('page.modelizer.settings.gitAddRemote.repository')"
       lazy-rules
-      data-cy="git-repository-input"
+      data-cy="repository-input"
```
[the remark this answers is not in the document]
Comment (justinehell) [reply src/components/form/GitAddRemoteForm.vue:20]: done

File: src/components/form/CreateFileForm.vue [hunk 23]
```diff
@@ -9,7 +9,7 @@
       filled
       :label="$t(`page.modelizer.fileExplorer.create.${isFolder ? 'folder' : 'file'}.input`)"
       lazy-rules
-      data-cy="create-file-input"
+      data-cy="name-input"
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply src/components/form/CreateFileForm.vue]: keep name-input

File: cypress/e2e/HomePage/create.project.feature [hunk 24]
```diff
@@ -2,11 +2,11 @@ Feature: Test homepage: project creation
 
   Background:
     Given I clear cache
-    And I visit the "/"
+    And  I visit the "/"
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply cypress/e2e/HomePage/create.project.feature]: missing quote correction (note to myself)

**Review by Zorin95670 — APPROVED** [review]

**Review by k-y0u — APPROVED** [review]

**Pull request merged**